### ISSUE TITLE
how to train OCR model?

### ISSUE THREAD

**Issue by linzhi123** [sergiomsilva/alpr-unconstrained#19]
Hello, do you have open source training OCR model code?

**Comment by sergiomsilva** [comment]
Hi @linzhi123 , sorry but we do not plan to publish this part anytime soon.

**Comment by danishansari** [comment]
@sergiomsilva Aren't you just using Darknet for OCR? If so, OCR can be trained separately using yolo, am i correct?

**Comment by linzhi123** [comment]
Yeah, I just want to train OCR alone. There's no training code. Yolo's training code is official, and his training code is modified. @danishansari

**Comment by linzhi123** [comment]
Do you have training code? Or do you know how to train? @danishansari

**Comment by danishansari** [comment]
Yes i am training it using YOLO-v2 with  the same network and weights that is there in this repository.
I am not able to get the loss reduced much, otherwise it showing good IOU(0.7) and Recall(0.96). 
Still working on it, haven't verified the result till now.

**Comment by linzhi123** [comment]
Can you share your code? @danishansari

**Comment by danishansari** [comment]
I used YOLO-v2 itself,  did not modified any of the code. I am just training for 36 classes instead of 35. So only config file has to be modified. So you can directly use Darknet for training and inference.

**Comment by linzhi123** [comment]
That's fine. Thank you!

**Comment by danishansari** [comment]
@linzhi123 please do share your experience and results(accuracy/map) after training.

**Comment by linzhi123** [comment]
ok @danishansari

**Comment by sergiomsilva** [comment]
I see you are interested in the training script, and I am sorry for not being able to share it with you guys at this moment because I'm really out of time. However I can give you some hints. The only modification we did in the original YOLO training algorithm was considering its data augmentation process. Basically, it reduces too much the image size, which is undesirable in this case, so we controlled it by performing small changes in the source code. We also included synthetically created data, as mentioned in the paper, but you may also achieve nice results without them.

**Comment by linzhi123** [comment]
Thanks very much @sergiomsilva

**Comment by Tomhouxin** [comment]
32/5000  
I want to ask how to use OCR training license plate data alone, I tried, but not so good @linzhi123

**Comment by danishansari** [comment]
@linzhi123 @Tomhouxin Can you please check out the memory usage of detector model, because when i am using it it's taking 10gbs of gpu memory. Is it really the memory requirement for this model?

**Comment by xinyuegtxy** [comment]
why not release the ocr training ????? em..............

**Comment by xinyuegtxy** [comment]
@linzhi123 hello, how do you train the ocr? And how do you prepare the dataset ?

**Comment by wanglaotou** [comment]
I want to know how to train ocr too. and could we change the code for two layer plate recognition?

**Comment by PhilipsKoshy** [comment]
> 
> 
> ok，I use tesseract4.0，and you？

could you share the details of how you used tesseract4.0? When I tried I get the error saying DPI information not present in the image.

**Comment by suvarnak** [comment]
> I want to know how to train ocr too. and could we change the code for two layer plate recognition?

yes. I wish to train it for plate with 2 rows
